### DISCUSSION TITLE
WebSocket session without a context manager?

### DISCUSSION THREAD

**Discussion by davidbrochart** [frankie567/httpx-ws#39] [Q&A] [answered]
[aconnect_ws](https://github.com/frankie567/httpx-ws/blob/d868919d5ab5d3e4514d554712e027e8a0a9abfb/httpx_ws/_api.py#L1168) is defined as an async context manager, that yields an [AsyncWebSocketSession](https://frankie567.github.io/httpx-ws/reference/httpx_ws/#httpx_ws.AsyncWebSocketSession), which can then be closed. But there is no way to easily create such a session without a context manager AFAIU (for manually closing it).
I currently do the following to get a `websocket`:
```py
from contextlib import AsyncExitStack
from httpx_ws import aconnect_ws

async def main():
    async with AsyncExitStack() as exit_stack:
        ws = aconnect_ws("my_url")
        websocket = await exit_stack.enter_async_context(ws)
        _exit_stack = exit_stack.pop_all()
        ...
        await _exit_stack.__aexit__()
```

**Comment by frankie567** [comment, marked as answer]
Didn't we discussed this some time ago ? ➡️ https://github.com/frankie567/httpx-ws/discussions/11 😄

**Reply by davidbrochart** [reply]
Ahah, looks like I have a <7 months memory :smile: 
Thanks!
